Back/Reciplastk.Gateway/Services/ShipmentTypeService.cs

```diff
@@ -0,0 +1,111 @@
+﻿using Reciplastk.Gateway.DataAccess;
+using Reciplastk.Gateway.Models;
+
+namespace Reciplastk.Gateway.Services
+{
+    public class ShipmentTypeService
+    {
+       private readonly ReciplastkContext db;
+       public ShipmentTypeService(ReciplastkContext db) 
+       {
+            this.db = db;
+       }
+        public HttpResponseModel ShowAllShipmentTypes()
+        {
+            var shipmentType = db.Shipmenttypes.Where(x => x.Isactive == true).ToList();
+            var response = new HttpResponseModel();
+            response.WasSuccessful = true;
+            response.Data = shipmentType;
+            return response;
+        }
+        public HttpResponseModel ShowShipmentType (int shipmenttypeid)
+        {
+            var response = new HttpResponseModel();
+            var shipmentType = db.Shipmenttypes.Where(x => x.Shipmenttypeid == shipmenttypeid && x.Isactive == true).FirstOrDefault();
+            if (shipmentType != null)
+            {
+                response.WasSuccessful = true;
+                response.Data = shipmentType;
+                response.StatusMessage = "El tipo de cargamento se encontro correctamente";
+            }
+            else
+            {
+                response.WasSuccessful = false;
+                response.StatusMessage = "No se encontro ningun tipo de cargamento con el id indicado";
+            }
+            return response;
+        }
+        private Shipmenttype FindShipmentById(int shipmenttypeid)
+        {
+            var shipmentType = db.Shipmenttypes.Where(x => x.Shipmenttypeid == shipmenttypeid && x.Isactive == true).FirstOrDefault();
+            return shipmentType;
+        }
+        public HttpResponseModel CreateShipmentType(ShipmentTypeViewModel shipmentTypeViewModel)
+        {
+            var response = new HttpResponseModel();
+            var shipmentType = FindShipmentById(shipmentTypeViewModel.shipmenttypeid);
+            if (shipmentType == null)
+            {
+                var newShipmentType = new Shipmenttype();
+                newShipmentType.Shipmenttypeid = shipmentTypeViewModel.shipmenttypeid;
+                newShipmentType.Name = shipmentTypeViewModel.name;
+                newShipmentType.Description = shipmentTypeViewModel.description;
+                newShipmentType.Creationdate = DateTime.Now;
+                newShipmentType.Isactive = true;
+                db.SaveChanges();
```

Comment:
Remember to use the add whenever you need to create a new row.
```suggestion
                db.Shipmenttypes.Add(newShipment); 
                db.SaveChanges();
```

Reply:
Solved